tests/functional/aws-node-sdk/test/bucket/putBucketReplication.js

```diff
@@ -0,0 +1,273 @@
+const assert = require('assert');
+const { errors } = require('arsenal');
+const { S3 } = require('aws-sdk');
+const { series } = require('async');
+
+const getConfig = require('../support/config');
+const replicationUtils = require('../../lib/utility/replication');
+
+const sourceBucket = 'source-bucket';
+const destinationBucket = 'destination-bucket';
+
+// Check for the expected error response code and status code.
+function assertError(err, expectedErr) {
+    if (expectedErr === null) {
+        assert.strictEqual(err, null, `expected no error but got '${err}'`);
+    } else {
+        assert.strictEqual(err.code, expectedErr, 'incorrect error response ' +
+            `code: should be '${expectedErr}' but got '${err.code}'`);
+        assert.strictEqual(err.statusCode, errors[expectedErr].code,
+            'incorrect error status code: should be 400 but got ' +
+            `'${err.statusCode}'`);
+    }
+}
+
+// Get parameters for putBucketReplication.
+function getReplicationParams(config) {
+    return {
+        Bucket: sourceBucket,
+        ReplicationConfiguration: config,
+    };
+}
+
+// Get parameters for putBucketVersioning.
+function getVersioningParams(status) {
+    return {
+        Bucket: sourceBucket,
+        VersioningConfiguration: {
+            Status: status,
+        },
+    };
+}
+
+// Get a complete replication configuration, or remove the specified property.
+const replicationConfig = {
+    Role: 'arn:partition:service::account-id:resourcetype/resource',
+    Rules: [
+        {
+            Destination: {
+                Bucket: `arn:aws:s3:::${destinationBucket}`,
+                StorageClass: 'STANDARD',
+            },
+            Prefix: 'test-prefix',
+            Status: 'Enabled',
+            ID: 'test-id',
+        },
+    ],
+};
+
+// Set the rules array of a configuration or a property of the first rule.
+function setConfigRules(val) {
+    const config = Object.assign({}, replicationConfig);
+    config.Rules = Array.isArray(val) ? val :
+        [Object.assign({}, config.Rules[0], val)];
+    return config;
+}
+
```

Comment:
Please add test that can't set putBucketReplication if request is from a different account (not bucket owner)